### PR TITLE
Implement BIP66

### PR DESCRIPTION
**Pull request by aalness**
See: https://github.com/bitcoin/bips/blob/master/bip-0066.mediawiki

Note: Actual DERSIG enforcement is done in bitcoin-ruby. This is merely the deployment handling code and related tests.

See also: https://github.com/bitcoin/bitcoin/pull/5713/files

### CODE REVIEW

**Comment by jplbigdata** [comment]
+1

**Comment by witoff** [comment]
+1

File: lib/toshi/processor.rb [hunk 1]
```diff
@@ -644,6 +644,14 @@ def accept_block(block, state)
         end
       end
 
+      # Reject block.nVersion=2 blocks when 95% (75% on testnet) of the network has upgraded:
+      if block.ver < 3
```
Comment (hudon) [inline lib/toshi/processor.rb]: I think it'd be a safer to put this check with the nVersion=1 check and refactor these constants out to be more like core https://github.com/bitcoin/bitcoin/blob/master/src/main.cpp#L2580

**Comment by hudon** [comment]
^minor comment, +1 otherwise

**Comment by aalness** [comment]
@hudon moved the version 2 check right after the version 1 check now. Refactor request of super majority code noted. I'm going to omit that specific refactor for now. There is a lot of refactoring I'd like to do now that I actually somewhat know Ruby. Also a more major refactor is needed since the Bitcoin Core code has diverged so much from what Toshi was ported from. The divergence is a bit dangerous and makes it increasingly possible we'll introduce a bug.

**Pull request merged**